src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1608,43 +1614,48 @@ else if (toUpdate.getContentAccessMode() != null) {
      * @return boolean status whether any updates are made or not.
      */
     private boolean updateEnvironment(ConsumerDTO updated, Consumer toUpdate) {
-        if (updated.getEnvironment() == null) {
+        List<EnvironmentDTO> environments = updated.getEnvironments();
+
+        if (environments == null) {
             return false;
         }
-
         boolean changesMade = false;
-        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
-        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
-        String validatedEnvId = null;
+        Map<String, String> validatedEnvIds = new HashMap<>();
 
-        if (environmentId != null) {
-            if (toUpdate.getEnvironmentId() == null || !toUpdate.getEnvironmentId().equals(environmentId)) {
+        if (environments != null) {
+            for (EnvironmentDTO environment : environments) {
+                String environmentId = environment == null ? null : environment.getId();
+                String environmentName = environment == null ? null : environment.getName();
 
-                if (!environmentCurator.exists(environmentId)) {
-                    throw new NotFoundException(i18n.tr(
-                        "Environment with ID \"{0}\" could not be found.", environmentId));
+                if (environmentId != null) {
+                    if (!environmentCurator.exists(environmentId)) {
+                        throw new NotFoundException(i18n.tr(
+                            "Environment with ID \"{0}\" could not be found.", environmentId));
+                    }
                 }
+                else if (environmentName != null) {
+                    environmentId = this.environmentCurator.
+                        getEnvironmentIdByName(toUpdate.getOwnerId(), environmentName);
 
-                validatedEnvId = environmentId;
-            }
-
-        }
-        else if (environmentName != null) {
-            String envId = this.environmentCurator.
-                getEnvironmentIdByName(toUpdate.getOwnerId(), environmentName);
+                    if (environmentId == null) {
+                        throw new NotFoundException(i18n.tr(
+                            "Environment with name \"{0}\" could not be found.", environmentName));
+                    }
+                }
+                else {
+                    throw new NotFoundException(i18n.tr("Environment could not be found."));
```

Comment:
This should be a bit more specific to indicate that it's not that we couldn't find the environment, but that there was no environment to find. Something along the lines of the environment being malformed.

Reply:
Done.